client/app/pages/queries/view.js

```diff
@@ -204,6 +204,25 @@ function QueryViewCtrl($scope, Events, $route, $routeParams, $location, $window,
     $scope.saveQuery(undefined, { is_draft: $scope.query.is_draft });
   };
 
+  $scope.saveAsDataset = () => {
```

Comment:
I tried to get this to work, but I needed to introduce another dependency and getting webpack to work started to get hairy. Not to mention the potential merge conflicts in the compilation config and `yarn.lock` when upgrading the underlying redash into our fork.

While it's nice, it felt like it wasn't worth the effort it was costing me for just this little piece of code.